game-app/game-core/src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -260,10 +260,13 @@ private TripleAFrame(
       final IGame game,
       final LocalPlayers players,
       final UiContext uiContext,
-      @Nullable final Chat chat) {
+      @Nullable final Chat chat,
+      final Runnable clientLeftGame) {
     super("TripleA - " + game.getData().getGameName());
     setDefaultCloseOperation(DISPOSE_ON_CLOSE);
 
+    this.clientLeftGame = clientLeftGame;
```

Comment:
`TripleAFrame` is literally interconnected with almost every UI element out there, so moving it to `:game-app:game-headed` was not feasible with reasonable effort.
That's why the method is passed in as a lambda.